tests/src/PatternAssertion/DescriptionListAssert.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the description list pattern.
+ */
+class DescriptionListAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'items' => [
+        [$this, 'assertItems'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+    $description_list = $crawler->filter('body > .bcl-description-list');
+    self::assertCount(1, $description_list);
+  }
+
+  /**
+   * Asserts the items of the pattern.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertItems(array $expected_items, Crawler $crawler): void {
+    // Initialize arrays to store labels, values, and icons.
+    $labels = [];
+    $values = [];
+    $dt_icons = [];
+    $dd_icons = [];
+
+    // Extract labels, values, and icons from expected items.
+    $this->extractLabelsValuesAndIcons($expected_items, $labels, $values, $dt_icons, $dd_icons);
+
+    // Assert labels.
+    $this->assertLabels($labels, $crawler);
+
+    // Assert icons in <dt> elements.
+    $this->assertIcons($dt_icons, 'dt', $crawler);
+
+    // Assert icons in <dd> elements.
+    $this->assertIcons($dd_icons, 'dd', $crawler);
+
+    // Assert values.
+    $this->assertValues($values, $crawler);
+  }
+
+  /**
+   * Extracts labels, values, and icons from expected items.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param array $labels
+   *   An array to store labels.
+   * @param array $values
+   *   An array to store values.
+   * @param array $dt_icons
+   *   An array to store icons for <dt> elements.
+   * @param array $dd_icons
+   *   An array to store icons for <dd> elements.
+   */
+  private function extractLabelsValuesAndIcons(array $expected_items, array &$labels, array &$values, array &$dt_icons, array &$dd_icons): void {
```

Comment:
this violates the single responsibility principle

Reply:
refactored.